view/src/components/foodbanks.js

```diff
@@ -0,0 +1,937 @@
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={[
+        "(",
+        /[1-9]/,
+        /\d/,
+        /\d/,
+        ")",
+        " ",
+        /\d/,
+        /\d/,
+        /\d/,
+        "-",
+        /\d/,
+        /\d/,
+        /\d/,
+        /\d/,
+      ]}
+      placeholderChar={"\u2000"}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+/**
+ * This class represents a Food Bank component, which is a sub-page of the
+ * home page where food bank objects are visualized, created, updated, edited,
+ * and deleted.
+ */
+class Foodbank extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // food bank states
+      foodbanks: "",
+      foodbankName: "",
+      location: "",
+      locationId: "",
+      hours: "",
+      foodbankId: "",
+      contactPhone: "(1  )    -    ",
+      contactName: "",
+      contactEmail: "",
+      forklift: false,
+      pallet: false,
+      loadingDock: false,
+      maxLoadSize: "",
+      refrigerationSpaceAvailable: "",
+      foodbankTags: [],
+      // page states
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEditClick = this.handleEditClick.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  /**
+   * Given an event, this function updates a state (the target of the event)
+   * with a new value
+   * @param event The event that is attempting to update a state
+   */
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /** Used to uncheck and check the form checkboxes */
+  handleChecked = (name) => (event) => {
+    this.setState({ ...this.state, [name]: event.target.checked });
+  };
+
+  /** Used to update tags in form */
+  onTagsChange = (event, values) => {
+    this.setState({
+      // food bank state
+      farmTags: values,
+    });
+  };
+
+  /** Used to update location from address autocomplete component */
+  handleLocation = (newValue) => {
+    if (newValue === null) {
+      return;
+    }
+    this.setState({
+      location: newValue.description,
+      locationId: newValue.place_id,
+    });
+  };
+
+  /** Returns the authentication token stored in local storage */
+  getAuth = () => {
+    authMiddleWare(this.props.history);
+    return localStorage.getItem("AuthToken");
+  };
+
+  /** Load in all of the current todos when the component has mounted */
+  componentDidMount() {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    axios
+      .get("/foodbanks")
+      .then((response) => {
+        this.setState({
+          foodbanks: response.data,
+          uiLoading: false,
+        });
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
console.log(err) is fine for debugging, but it would be a good idea to let the user know that something went wrong, which console.log won't.  Maybe in another PR you can swap all of these out with some code that shows a toast/snackbar indicating that an error has occurred.

Reply:
A few commits down the line I add alert functionality which uses exactly what you suggest to visualize every call to console.log or to show something has occurred successfully. I am waiting to get a few more files merged in before I have a PR with the alert aparatus.

Reply:
Sounds good, thanks!